src/Tribe/Events/Events_Service_Provider.php

```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace Tribe\Tickets\Events;
+
+class Events_Service_Provider extends \tad_DI52_ServiceProvider {
+
+	/**
+	 * Register classes in the container that Event Tickets uses
+	 * to manage Events.
+	 *
+	 * @since TBD
+	 */
+	public function register() {
```

Comment:
Is there a condition where this service provider, and with it all the actions and filters, should not register at all?
If so, then check that condition here and bailout of any other registration if that condition is not satisfied.

Reply:
I could think of five contexts where this could be called:
- is_admin()
- wp_doing_cron()
- wp_doing_ajax()
- wp_using_themes()
- CLI

I could probably bail on Cron, but I'm not sure about the others.

Reply:
Would this be called if TEC Main class does not exist?

Reply:
Hmm, this makes sense. And actually, this filter is probably wrong: https://github.com/moderntribe/event-tickets/pull/1633/files#diff-0bb16ce448c6735634b9bb57174aa718R37

With Event Tickets, an Event can be of any post type, not just `tribe_events`. Is that right, @sc0ttkclark ?

Reply:
Fixed https://github.com/moderntribe/event-tickets/pull/1633/commits/3c5f255c1e90be2ca0c889a6f8a14d1a1f673edb